### PR TITLE
Add temperature attribute to BatteryData

### LINKED ISSUE #37: BMS temperature data support
### What's needed?

Battery temperature data is needed for lynch in order to decide whether to (dis)charge early within a 15min window.  Lynch will use the **max** temperature of all batteries in the microgrid. Hence only one value(highest temperature out of all batteries) is necessary for this usecase.

### Proposed solution

Either one value outputted which gives the max temperature or temperature of each battery. Both solutions work for this usecase.

### Use cases

Peak shaving (GMM)

### Alternatives and workarounds

_No response_

### Additional context

Even though for peak shaving only the max temperature is required, usecases in the future might need separate temperatures for each battery.

### PR DESCRIPTION
**Pull request by jkozlowicz**

### CODE REVIEW

File: src/frequenz/sdk/microgrid/component_data.py [hunk 1]
```diff
@@ -127,6 +127,8 @@ class BatteryData(ComponentData):
         power_upper_bound: the maximum charge power, in Watts, represented in the
             passive sign convention. This will be a positive number, or zero if no
             charging is possible.
+        temperature: The aggregated values of all the temperature measurements of a
+            battery, in Celsius (°C).
```
Comment (ela-kotulska-frequenz) [inline src/frequenz/sdk/microgrid/component_data.py:131]: Temperature of the battery, in Celsius (°C).

File: src/frequenz/sdk/microgrid/component_data.py [hunk 2]
```diff
@@ -155,6 +158,7 @@ def from_proto(cls, raw: microgrid_pb.ComponentData) -> BatteryData:
             capacity=raw.battery.properties.capacity,
             power_lower_bound=raw.battery.data.dc.power.system_bounds.lower,
             power_upper_bound=raw.battery.data.dc.power.system_bounds.upper,
+            temperature=raw.battery.data.temperature.max,
```
Comment (leandro-lucarella-frequenz) [inline src/frequenz/sdk/microgrid/component_data.py:161]: This should probably be `temperature_max`?

**Review by leandro-lucarella-frequenz — COMMENTED** [review]
@idlir-shkurti-frequenz @sahas-subramanian-frequenz maybe you can give this a review too?

We could merge it so it is shipped in the next release...

**Comment by sahas-subramanian-frequenz** [comment]
Closing in favour of https://github.com/frequenz-floss/frequenz-sdk-python/pull/75

**Pull request closed without merging**